src/dash-table/components/Table/index.tsx

```diff
@@ -176,15 +205,27 @@ export default class Table extends Component<PropsWithDefaultsAndDerived> {
             return;
         }
 
-        setTimeout(() => setProps(newProps), 0);
+        setTimeout(() => controlledSetProps(newProps), 0);
     }
 
-    private get setProps() {
+    private get controlledSetProps() {
         return this.__setProps(this.props.setProps);
     }
 
+    private get controlledSetState() {
+        return this.__setState();
+    }
+
     private readonly __setProps = memoizeOne((setProps?: SetProps) => {
         return setProps ? (newProps: any) => {
+            /*#if DEV*/
+            const props: any = this.props;
+            R.forEach(
+                key => props[key] === newProps[key] && Logger.fatal(`Updated prop ${key} was mutated`),
```

Comment:
Some additional sanity here, make sure we are not sending a mutated version of an existing props property -- since some of the comps are pure, this prevents updates from happening correctly